### ISSUE TITLE
Cache No Longer Supported, FAQ Incorrect

### ISSUE THREAD

**Issue by clukewatson** [sergiocorreia/reghdfe#197]
Howdy,

First, I love reghdfe as a research tool; thank you. 
Machine: 2018 MBP, Stata 13.1 (with updates), reghdfe (*! version 5.6.8 03mar2018)

Issue: Is "cache()" no longer supported? I wish to use the same controls + FEs with different dependent variables, but this seems not to work currently.

Example of Issue:
sysuse auto
reghdfe price weight length gear_ratio, absorb(turn rep) cache(save)
reghdfe gear_ratio weight length, a(turn rep) cache(use)

Thank you for your time,
Best,
Luke

**Comment by sergiocorreia** [comment]
Yep, you are right. When I ported reghdfe to fully use Mata (and ftools), some things were not porting (mostly the cache option). Partly because it was very complicated, and partly because the new version is fast enough that I don't have much of a need.

That said, you *can* still access cache if you run reghdfe with the `old` option.

**Comment by reifjulian** [comment]
I installed the latest developer's version of `reghdfe` (as of September 14, 2023):
```
cap ado uninstall ftools
net install ftools, from("https://raw.githubusercontent.com/sergiocorreia/ftools/master/src/")

cap ado uninstall reghdfe
net install reghdfe, from("https://raw.githubusercontent.com/sergiocorreia/reghdfe/master/src/")

mata mata mlib index
```
I then tried using the `cache` option, but got an error when trying the example from the [faq](http://scorreia.com/software/reghdfe/faq.html), regardless of whether or not I specify the `old` option :

```
sysuse auto, clear
reghdfe price weight length, absorb(turn trunk) cache(save) old
reghdfe price weight length, absorb(turn trunk) cache(save)
```

![image](https://github.com/sergiocorreia/reghdfe/assets/42722121/8f0c4cb9-c79e-41be-8060-932c7a8b3160)

Even with speedy `reghdfe`, my regression takes up to a day or more on Stata-MP, and I have many outcomes, so any potential speed-ups are definitely appreciated on my end :)

**Comment by sergiocorreia** [comment]
Hi Julian,

The `cache` option existed until reghdfe v3 (see `help reghdfe3`) and can be used through the `version()` option of reghdfe:

```stata
sysuse auto
reghdfe price weight length , a(turn) version(3) cache(save)
reghdfe price weight length , a(turn) version(3) cache(use)
```

I removed it from reghdfe because it was making the code quite messy, and in the cases I needed it most I could achieve better speedups by directly calling the reghdfe mata internals (see `help reghdfe_programming`)

I do need to update the online help though, but between policy and research sadly I haven't prioritized it enough!

**Comment by reifjulian** [comment]
Got it, thanks for the info, Sergio! It seems like the `noregress` option, which appears to use the new version of `reghdfe` with Mata/ftools, can be used to estimate regressions with multiple outcomes, although it wasn't clear to me how to run subsequent regressions after specifying `noregress`, or whether the std errors would still be corrected for the partialling out in the first step. If it's possible to speedily estimate multiple regressions using `noregress` --and if you ever have time (!) to put together a simple example showing how--I think it would be very valuable. This situation with multiple outcomes arises for a lot of researchers, who in some cases turn to `fixest` in R when the regressions are big.
